### PR TITLE
Test structure

### PR DESCRIPTION
**Pull request by JohnSchramm**
Took a first stab at applying the WIP structure to a full army (Slaanesh).

### CODE REVIEW

**Comment by exonian** [comment]
This is amazing work

File: src/factions/slaanesh/spells.ts [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+    effects: [
+      {
+        name: `Phantasmagoria`,
+        desc: `Casting value of 7. Pick 1 enemy unit within 18" of the caster that is visible. Roll 6 dice and for each 5+ until your next hero phase subtract 1 from that unit's bravery characteristic (to a minimum of 1).`,
+        when: [HERO_PHASE],
+        spell: true,
+      },
+      {
+        name: `Phantasmagoria`,
+        desc: `If active, subtract 1 from that unit's bravery characteristic (to a minimum of 1).`,
```
Comment (daviseford) [inline src/factions/slaanesh/spells.ts:64]: While we're re-inventing everything... I've always wondered if we should have a tag for like... `spell_active_effect` for  things like this.

File: src/factions/slaanesh/subfactions.ts [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+
+    // Endless Spells
+    endless: EndlessSpells,
+
+    // Scenery
+    scenery: Scenery,
+
+    // Units
+    units: [
+      Units,
```
Comment (daviseford) [inline src/factions/slaanesh/subfactions.ts:131]: So I think we need to turn Units into an array, rather than an object

**Comment by daviseford** [comment]
Honestly @JohnSchramm you are the most amazing person <3

**Pull request merged**